### PR TITLE
733: case deadline search and edit

### PR DESCRIPTION
**Pull request by kkoskelin**
Screen Shot 2020-08-13 at 12 11 45 PM

### CODE REVIEW

File: web-client/src/views/CaseDeadlines/DatePickerComponent.jsx [hunk 1]
```diff
@@ -37,10 +40,10 @@ export const DatePickerComponent = ({
   }, [value]);
 
   useEffect(() => {
+    const input = document.getElementById(`${name}-date`);
```
Comment (matthopson) [inline web-client/src/views/CaseDeadlines/DatePickerComponent.jsx:43]: Should we use a ref for this instead? I think diving into the DOM this way is a bit of an anti-pattern.

File: web-client/src/views/CaseDeadlines/DatePickerComponent.jsx [hunk 2]
```diff
@@ -86,13 +89,10 @@ export const DatePickerComponent = ({
       <div className="usa-hint" id={`${name}-date-hint`}>
         MM/DD/YYYY
       </div>
-      <div className="usa-date-picker">
+      <div className="usa-date-picker" data-default-value={defaultValue}>
```
Comment (matthopson) [inline web-client/src/views/CaseDeadlines/DatePickerComponent.jsx:92]: Similar to the previous comment - I'm surprised this isn't working as intended. Could you elaborate on the issue a little?

File: web-client/src/views/CaseDeadlines/DatePickerComponent.jsx (hunk 1 above)
Comment (kkoskelin) [reply web-client/src/views/CaseDeadlines/DatePickerComponent.jsx:43]: I'm not arguing with you on that point, but at this time I'm following a pattern that makes this thing *work*.  I agree that this needs to be done better, and that will include quite possibly submitting one or more PRs to the USWDS project.

File: web-client/src/views/CaseDeadlines/DatePickerComponent.jsx (hunk 2 above)
Comment (kkoskelin) [reply web-client/src/views/CaseDeadlines/DatePickerComponent.jsx:92]: looking at the USWDS `date-picker.js`, it became evident that the react-ish `defaultValue` on the input itself would not be accessible by the module's vanilla js post-render.  It also became evident that the module's JS was indeed looking for `data-default-value` on the element matching `.usa-date-picker`.  I know.  Don't shoot the messenger.

Bottom line is that the modal dialog can open up AND the input shows the default value correctly.

**Comment by kkoskelin** [comment]
Also: the page went completely blank with a "cannot assign value to null" or somesuch when attempting to edit a deadline prior to this fix.  Which is no bueno.

File: web-client/src/views/CaseDeadlines/DatePickerComponent.jsx (hunk 2 above)
Comment (matthopson) [reply web-client/src/views/CaseDeadlines/DatePickerComponent.jsx:92]: Yeah, perhaps we should share this feedback with USWDS.

**Review by matthopson — APPROVED** [review]

File: web-client/src/views/CaseDeadlines/DatePickerComponent.jsx (hunk 2 above)
Comment (matthopson) [reply web-client/src/views/CaseDeadlines/DatePickerComponent.jsx:92]: Could we make a devex task to try and revisit this later (just so we don't lose it).

Comment (kkoskelin) [reply web-client/src/views/CaseDeadlines/DatePickerComponent.jsx:92]: https://trello.com/c/6PkJl8WC/736-address-uswds-datepicker-issues

**Pull request merged**